src/modeler/AutoRest.Swagger/Validation/BooleanPropertyNotRecommended.cs

```diff
@@ -42,25 +42,27 @@ public class BooleanPropertyNotRecommended : TypedRule<Dictionary<string, Schema
         /// </summary>
         /// <param name="definitions">Operation Definition to validate</param>
         /// <returns>true if there are no propeties of type boolean, false otherwise.</returns>
-        public override bool IsValid(Dictionary<string, Schema> definitions, RuleContext context, out object[] formatParameters)
+        public override IEnumerable<ValidationMessage> GetValidationMessages(SwaggerObject entity, RuleContext context)
         {
-            formatParameters = null;
-            List<string> booleanProperties = new List<string>();
-            foreach (KeyValuePair<string, Schema> definition in definitions)
+            if (entity.Type.ToString().ToLower().Equals("boolean"))
```

Comment:
minor: we should check `entity.Type` with `DataType.Boolean` check on Type object itself instead of string